### DISCUSSION TITLE
Am I correct with how Flux (should) work?

### DISCUSSION THREAD

**Discussion by Igorgro** [fluxcd/flux2#3841] [Q&A] [answered]
I have Installed Flux on my Kubernetes cluster and set up it to watch some git repository using `GitRepository` manifest. Now I want to deploy something to cluster with Kustomize and Flux. I created a folder in that repository which contains `kustomization.yaml` and some patch files and pushed it. Now I want the Flux to apply this kustomization. To achieve that I created `kustomize.toolkit.fluxcd.io/v1 Kustomization` manifest (from [here](https://fluxcd.io/flux/components/kustomize/kustomization/#writing-a-kustomization-spec)) which references that `GitRepository` resource and the kustomization directory in that repository. When I just commit and push this Flux Kustomization manifest to that repository, then nothing happens. But when I manually apply this manifest using `kubectl`, Flux pulls the data from the repo and successfuly deply the thing defined in kustomization. My question is the following: is that correct behaviour of Flux? Should it automatically apply pushed Kustomization manifest (not kustomization.yml!) or not?

**Comment by makkes** [comment]
Flux doesn't automatically create any resources based exclusively on the existence of a `GitRepository` in the cluster. A `GitRepository` merely points Flux to a Git repository which Flux then provides as an artifact to other components (e.g. [kustomize-controller](https://fluxcd.io/flux/components/kustomize/)).

The best way for you to get started would be to [bootstrap Flux](https://fluxcd.io/flux/installation/#bootstrap) using `flux bootstrap` which creates a `GitRepository` and a `Kustomization`. From there please follow the various guides on https://fluxcd.io/flux/.

**Reply by Igorgro** [reply]
Yes, I already bootstrapped Flux but thought that it will automatically apply all manifests from configured repository - this is what "GitOps" means

**Reply by makkes** [reply, marked as answer]
The `Kustomization` manifest that's generated as part of the bootstrap process is the root of your GitOps deployment. From there you can go ahead and e.g. add manifests to the `flux-system/kustomization.yaml`. Flux will only reconcile what it is explicitly pointed to.
